test/indexing.jl

```diff
@@ -57,6 +57,12 @@ module TestIndexing
         end
     end
 
+    # ensure that the NullableArray returned by getindex maintains its parent field
+    str = "something"
+    x = NullableArray([pointer(str), pointer(str, 5)], [false, false], Vector{UInt8}(str))
+    @test isequal(x[1:2].parent, x.parent)
```

Comment:
Thanks. I've made that change, and the corresponding change to test/primitives.jl as well.